src/lexer/token.ts

```diff
@@ -35,13 +35,16 @@ export interface Token {
   raw: string; // The raw original text that was matched
   text: string; // Cleaned up text e.g. keyword converted to uppercase and extra spaces removed
   key?: string;
+  index: number; // index of the token in the whole query string
+  line?: number; // line number that the token appears on
+  col?: number; // column within the line on which the token appears
```

Comment:
When dealing with line and column numbers, one always needs to specify whether these are zero- or one-based. i.e. is the first line number `0` or `1`. 

Usually one wants to use one-based indexing when displaying the info to user, while for internal usage, zero-based indexing is better for doing calculations with.